--- a/tty-keys.c
+++ b/tty-keys.c
@@ -1663,6 +1663,8 @@
 		tty_default_features(features, "foot", 0);
 	else if (strncmp(tmp, "WezTerm ", 7) == 0)
 		tty_default_features(features, "WezTerm", 0);
+	else if (strncmp(tmp, "ghostty ", 8) == 0)
+		tty_default_features(features, "ghostty", 0);
 	log_debug("%s: received extended DA %.*s", c->name, (int)*size, buf);
 
 	free(c->term_type);
--- a/tty-features.c
+++ b/tty-features.c
@@ -543,6 +543,18 @@
 			      "focus,"
 			      "usstyle"
 		},
+		{ .name = "ghostty",
+		  .features = TTY_FEATURES_BASE_MODERN_XTERM ","
+			      "ccolour,"
+			      "cstyle,"
+			      "extkeys,"
+			      "focus,"
+			      "hyperlinks,"
+			      "osc7,"
+			      "sync,"
+			      "usstyle,"
+			      "progressbar"
+		},
 		{ .name = "XTerm",
 		  /*
 		   * xterm also supports DECSLRM and DECFRA, but they can be
